frontend/src/providers/QueryBuilder.tsx

```diff
@@ -95,7 +95,17 @@ export function QueryBuilderProvider({
 	const urlQuery = useUrlQuery();
 	const history = useHistory();
 	const location = useLocation();
-	const currentPathnameRef = useRef<string | null>(null);
+	/**
+	 // set location.pathname as the initial value for currentPathnameRef, absense of this causes `stagedQuery && location.pathname !== currentPathnameRef.current` to be truthy (since currentPathnameRef.current is null), and therefore causes duplicate triggering of initQueryBuilderData(compositeQueryParam); in the useEffect of line ~784
```

Comment:
let's add this as PR description. Inline comment will pollute the provider here

Reply:
Sure, :+1: